tests/lib/testing_classes.py

```diff
@@ -347,15 +372,36 @@ def publish_finished(self, dbcon, launched_app, download_test_data,
         time_start = time.time()
         timeout = timeout or self.TIMEOUT
         timeout = float(timeout)
+
+        if keep_app_open:
+            timeout = 100000
+
+        new_lines = []
         while launched_app.poll() is None:
+            with open(stdout_path, "r") as f:
+                for line in f.readlines():
+                    if line not in new_lines:
+                        new_lines.append(line)
+                        print(line.strip())
```

Comment:
This is due to MayaPy outputting stdout to a file, so the console is blank. By reading in the output file, we can get some interactive inspection.

Reply:
Why to store them to `new_lines` attribute?

Reply:
So we can query whether the line has been printed/stored already. This is so we dont print repeat lines. Open to suggestions.

Reply:
I guess a better solution would be to store the previous line and compare against the new lines?

Reply:
I think you should store the index where you were last time, and go to that index on next loop.

This is untested prototype. Probably would require some tweaks.
```python
s_idx = 0
while launched_app.poll() is None:
    with open(stdout_path, "r") as stream:
        stream.seek(s_idx)
        for line in stream.readlines():
            print(line.strip())
        s_idx = stream.tell()
```

This would maybe print all remaining output that was not printed because process ended.
```python
def print_stream_from_idx(path, idx):
    with open(path, "r") as stream:
        stream.seek(idx)
        for line in stream.readlines():
            print(line.strip())
        new_idx = stream.tell()
    return new_idx


s_idx = 0
while launched_app.poll() is None:
    # Live output while process is alive
    s_idx = print_stream_from_idx(stdout_path, s_idx)
# Remaining output
print_stream_from_idx(stdout_path, s_idx)
```

Reply:
Changing the flag behaviour to only operate with UI, so this is redundant now. Thought it was overcomplicating things anyways.